contracts/validator-manager/tests/ValidatorMessagesTests.t.sol

```diff
@@ -23,13 +25,261 @@ contract ValidatorMessagesTest is Test {
     // solhint-disable-next-line var-name-mixedcase
     PChainOwner public DEFAULT_P_CHAIN_OWNER;
+    address public constant DEFAULT_OWNER = 0x1234567812345678123456781234567812345678;
 
     function setUp() public {
         address[] memory addresses = new address[](1);
-        addresses[0] = 0x1234567812345678123456781234567812345678;
+        addresses[0] = DEFAULT_OWNER;
         DEFAULT_P_CHAIN_OWNER = PChainOwner({threshold: 1, addresses: addresses});
     }
 
+    function testSubnetConversionMessageInvalidInputLength() public {
+        bytes memory packed =
+            ValidatorMessages.packSubnetToL1ConversionMessage(DEFAULT_SUBNET_CONVERSION_ID);
+        // Invalid length
+        bytes memory invalidPacked = new bytes(packed.length - 1);
+        for (uint256 i = 0; i < packed.length - 1; i++) {
+            invalidPacked[i] = packed[i];
+        }
+        vm.expectRevert(
+            abi.encodeWithSelector(ValidatorMessages.InvalidMessageLength.selector, 37, 38)
+        );
+        ValidatorMessages.unpackSubnetToL1ConversionMessage(invalidPacked);
+    }
+
+    function testSubnetConversionMessageInvalidCodecID() public {
+        bytes memory packed =
+            ValidatorMessages.packSubnetToL1ConversionMessage(DEFAULT_SUBNET_CONVERSION_ID);
+
+        // Invalid codec ID
+        bytes memory invalidPacked2 = packed;
+        invalidPacked2[1] = 0x01;
+        vm.expectRevert(
+            abi.encodeWithSelector(ValidatorMessages.InvalidCodecID.selector, uint32(1))
+        );
+        ValidatorMessages.unpackSubnetToL1ConversionMessage(invalidPacked2);
+    }
+
+    function testSubnetConversionMessageInvalidTypeID() public {
+        bytes memory packed =
+            ValidatorMessages.packSubnetToL1ConversionMessage(DEFAULT_SUBNET_CONVERSION_ID);
+        // Invalid message type
+        bytes memory invalidPacked3 = packed;
+        invalidPacked3[5] = 0x01;
+        vm.expectRevert(ValidatorMessages.InvalidMessageType.selector);
+        ValidatorMessages.unpackSubnetToL1ConversionMessage(invalidPacked3);
+    }
+
+    function testRegisterSubnetValidatorMessageInvalidBLSKey() public {
+        vm.expectRevert(ValidatorMessages.InvalidBLSPublicKey.selector);
+        // 47 bytes
+        bytes memory invalidBLSKey = bytes(
+            hex"3456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678"
+        );
+        ValidatorMessages.packRegisterL1ValidatorMessage(
+            ValidatorMessages.ValidationPeriod({
+                subnetID: DEFAULT_SUBNET_ID,
+                nodeID: DEFAULT_NODE_ID,
+                registrationExpiry: DEFAULT_EXPIRY,
+                blsPublicKey: invalidBLSKey,
+                remainingBalanceOwner: DEFAULT_P_CHAIN_OWNER,
+                disableOwner: DEFAULT_P_CHAIN_OWNER,
+                weight: DEFAULT_WEIGHT
+            })
+        );
+    }
+
+    function testRegisterSubnetValidatorMessageInvalidInputLength() public {
```

Comment:
Should we change the name here? It looks like this test checks 3 different invalid cases and not just `InvalidInputLength`

Reply:
Yeah I think the two tests below got copied into this test accidentally on a merge. All fixed now.